patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,143 @@
+    - Set up [Single Sign-On][single-sign-on] against your organization’s identity provider. This will automatically revoke access in case somebody leaves the organization.
+    - regular rotation of access token
+    - 2FA authentication enforced
+    - encourage the use of signed commits
+- Monitoring the source code management system for suspicious access, e.g.:
+    - access from new accounts or new devices
+    - downloading source code from new devices
+    - downloading a great amount of source code in the monitoring period.
+
+### Prevent sensitive information in shared repositories
+
+Build up agreed security requirement of InnerSource, such as:
+
+- No hard-coded usernames, passwords or other access/security tokens in a repository.
+- No client information in a repository.
+- Business critical algorithms may not be placed in SHARED repositories.
+- Code from SHARED repositories will not be distributed directly to Production.
+- Leverage secrets scanning tools to scan the target repositories (including code, test cases, and helper scripts) for confidential data such as accounts, passwords, access tokens, keys, and other sensitive data.
+- Keep repositories CLOSED, but expose metadata about the projects (i.e. through an [InnerSource Portal][innersource-portal]), and create some kind of access request workflow. This way you could still give people access to the code, but not open it to everyone by default. *(Note: The full model is described under **Extension: An Additional Sharing Level**)*
+
+### Extension: An Additional Sharing Level
+
+In some cases, introducing additional sharing levels might be appropriate. Use cases include semi-sensitive code that can be shared with larger groups of people, but for example not with all interns and contractors. Or not with people with certain nationalities or working in a specific country, in order to comply with [export control][export-control] regulations. Or legacy repositories that are not (yet) ready to be made available as SHARED. Without the additional level, (too) many repositories will end up as CLOSED in an existing organization, as all businesses will considering their code 'special', requiring extra protection and/or 'not ready for opening up company wide yet (e.g. hard-coded secrets, but also technical debt, lacking documentation).
+
+And given findability is also important, adding the additional level that facilitates findability, but still provides a light-weight access gate could be an appropriate way to prevent too many CLOSED repositories.
+
+As a concrete example, which allows repositories to be discovered, even though there is no company-wide access, consider the following sharing levels:
+
+- **PUBLIC** - open source: findable and accessible for all SW developers in the world
+- **INTERNAL** - inner source: findable and accessible for all SW developers in the company
```

Comment:
@bartgolsteijn what is called INTERNAL here, is the same as what is called SHARED in line 52. I am wondering if we might confuse the reader by renaming it?

What if we would keep all names the same, and just add RESTRICTED?

Reply:
Fine with me.

Context: these are the names used my company. 
Also depends a bit on how you view it: introducing an additional level between SHARED and CLOSED or splitting the SHARED level in more and less shared. 

Given I am biassed here, I prefer others decide.

Reply:
Let's leave it as is for now then. 

I will leave this thread open for future reference.